python_modules/dagster/dagster_tests/core_tests/test_object_store.py

```diff
@@ -52,3 +94,35 @@ def test_s3_object_store():
 
     finally:
         object_store.rm_object(context, ['true'])
+
+
+@aws
+@nettest
+def test_s3_object_store_with_custom_serializer():
+    run_id = str(uuid.uuid4())
+
+    # FIXME need a dedicated test bucket
+    object_store = S3ObjectStore(run_id=run_id, s3_bucket='dagster-airflow-scratch')
+
+    try:
+        with yield_pipeline_execution_context(
+            PipelineDefinition([]), {}, RunConfig(run_id=run_id)
+        ) as context:
+            object_store.set_object('foo', context, LowercaseString.inst(), ['foo'])
+
+        assert (
+            object_store.s3.get_object(
+                Bucket=object_store.bucket, Key='/'.join([object_store.root] + ['foo'])
+            )['Body']
+            .read()
+            .decode('utf-8')
+            == 'FOO'
+        )
+
+        assert object_store.has_object(context, ['foo'])
+        assert object_store.get_object(context, LowercaseString.inst(), ['foo']) == 'foo'
+    finally:
+        try:
+            shutil.rmtree(object_store.root)
```

Comment:
very confused by this

Reply:
this is just test case cleanup

Reply:
i think you fixed this by making it rm_object actually